volatility3/framework/layers/qemu.py

```diff
@@ -33,15 +37,47 @@ class QemuSuspendLayer(segmented.NonLinearlySegmentedLayer):
     SEGMENT_FLAG_XBZRLE = 0x40
     SEGMENT_FLAG_HOOK = 0x80
 
+    # See https://qemu.readthedocs.io/en/latest/devel/memory.html for more info
+    #
+    # At least the following values could occur for devices using > 3-4 GB RAM:
+    # +--------------------------------+--------------------------------+------------+-------------+
+    # | Architecture                   | Reference Code                 | Hole Start | Hole End    |
+    # +--------------------------------+--------------------------------+------------+-------------+
+    # | PC i440FX + PIIX "New Default" | qemu/hw/i386/pc_piix.c:98      | 0xc0000000 | 0x100000000 |
+    # | PC i440FX + PIIX "Old Default" | qemu/hw/i386/pc_piix.c:98      | 0xe0000000 | 0x100000000 |
+    # | PC Q35 + ICH9                  | qemu/hw/i386/pc_q35.c:141      | 0x80000000 | 0x100000000 |
+    # | MicroVM                        | qemu/hw/i386/microvm.c:291     | 0xc0000000 | 0x100000000 |
+    # | Xen                            | qemu/hw/i386/xen/xen-hvm.c:248 | 0xf0000000 | 0x100000000 |
+    # +--------------------------------+--------------------------------+------------+-------------+
+    #
+    # For now, we assume that the parameter max-ram-below-4g is not set, since this parameter influences the size
+    # and location of the memory gap. Deviating hole sizes could eventually be detected for Linux by e.g. scanning
+    # for dmesg entries with a regex like rb'\[mem (0x[0-9a-f]{4,10})-0x[0-9a-f]{4,10}\] available for PCI devices'
+
+    distro_re = r"(artful|eoan|rhel[\d\.]+)"
```

Comment:
We could maybe replace this with `r"(\w+|rhel[\d\.]+)"` to support all distro names?

Reply:
Yeah, I thought it might be better to get bug reports when one comes out we don't support, but I guess it won't really change how we handle it.

Reply:
I'm not sure the version number won't potentially follow other distros, so I've gone for:
```distro_re = r"(\w+[\d\.]?)"```